src/main/update/player-update.ts

```diff
@@ -0,0 +1,62 @@
+import { DownloadItem, app } from "electron";
+import { download, Options as ElectronDLOptions } from "electron-dl";
+import { IDownloadProgress } from "src/interfaces/ipc";
+import { DownloadBinaryFailedError } from "../exceptions/download-binary-failed";
+import path from "path";
+import { IUpdateOptions } from "./launcher-update";
+import { macExtract, winExtract } from "./extract";
+
+const playerTempPath = path.join(app.getPath("temp"), "player");
+const extractPath = path.join(app.getPath("userData"), "player");
+
+export async function playerUpdate(
+  downloadUrl: string,
+  listeners: IUpdateOptions
+) {
+  const win = listeners.getWindow();
+
+  if (win === null) {
+    console.log("Stop update process because win is null.");
+    return;
+  }
+
+  win?.webContents.send("update download started");
```

Comment:
I don't feel okay with reusing the update events for player updates again. As far as my knowledge of the update events, the current `updateMachine.ts` assumes all the steps are irreversible (i.e. you can't go back from extract to download)

Reply:
Thanks you, then... I'll solve to combine window send event